src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -92,10 +112,17 @@ export function TopNavMenu(props: TopNavMenuProps): ReactElement | null {
   }
 
   function renderLayout() {
+    const { setMenuMountPoint } = props;
     const className = classNames('kbnTopNavMenu', props.className);
     return (
       <span className="kbnTopNavMenu__wrapper">
-        {renderMenu(className)}
+        {setMenuMountPoint ? (
+          <MountPointPortal setMountPoint={setMenuMountPoint}>
+            {renderMenu(className)}
+          </MountPointPortal>
+        ) : (
+          renderMenu(className)
+        )}
         {renderSearchBar()}
       </span>
```

Comment:
I wonder if that would be preferable, when we are rendering via portal (`setMenuMountPoint` is defined), to duplicate the wrapper and do something like 

```tsx
    if (setMenuMountPoint) {
      return (
        <>
          <MountPointPortal setMountPoint={setMenuMountPoint}>
            <span className="kbnTopNavMenu__wrapper">{renderMenu(className)}</span>
          </MountPointPortal>
          <span className="kbnTopNavMenu__wrapper">{renderSearchBar()}</span>
        </>
      );
    } else {
      return (
        <span className="kbnTopNavMenu__wrapper">
          {renderMenu(className)}
          {renderSearchBar()}
        </span>
      );
    }
```

wdyt?

Reply:
I prefer the duplicated code for readability 👍 
May want to extract the className of the span into a variable that can be reused to prevent inconsistency bugs from changes to the classname.